internal/metrics/url_response.go

```diff
@@ -0,0 +1,57 @@
+package metrics
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	ctrlmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	externalURLs = []string{"https://httpstat.us/503", "https://httpstat.us/200"}
+
+	availability = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "reference_addon_sample_availability",
+			Help: "external url availability.",
+		},
+		[]string{"url"},
+	)
+	responseTime = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "reference_addon_sample_response_time",
+			Help: "external url response time taken.",
+		},
+		[]string{"url"},
+	)
+)
+
+// RegisterMetrics must register metrics in given registry collector
+func RegisterMetrics() {
+	ctrlmetrics.Registry.MustRegister(availability)
+	ctrlmetrics.Registry.MustRegister(responseTime)
+}
+
+func RequestSampleResponseData() {
+	for _, externalURL := range externalURLs {
```

Comment:
If an `externalURL` has a problem, but the `reference-addon` is healthy, the `availability` metric will be set to `0`, although the `reference-addon` is healthy.